Watchman.Discord/Areas/Administration/Controllers/AdministrationController.cs

```diff
@@ -87,34 +87,68 @@ public async Task ReadUserMessages(DiscordRequest request, Contexts contexts)
 
-            var roleName = args[0].Value;
-            var toSetAsSafe = args[1].Value == "safe";
-
-            var serverRole = _usersRolesService.GetRoleByName(roleName, contexts.Server);
-            if (serverRole == null)
+            if (args.HasDuplicates())
             {
-                throw new RoleNotFoundException(roleName);
+                throw new ArgumentsDuplicatedException();
             }
 
-            if (toSetAsSafe)
+            var lastArgument = args.Last().Value.ToLower();
+
+            var argumentIsNotCorrect = lastArgument != "safe" && lastArgument != "unsafe";
+
+            if (argumentIsNotCorrect)
             {
-                var command = new SetRoleAsSafeCommand(roleName, contexts.Server.Id);
-                await _commandBus.ExecuteAsync(command);
+                throw new NotEnoughArgumentsException();
             }
-            else
+
+            var isSafe = lastArgument == "safe" ? true : false;
+            var safeRolesQuery = new GetDiscordServerSafeRolesQuery(contexts.Server.Id);
+
+            var safeRoles = this._queryBus.Execute(safeRolesQuery).SafeRoles;
+            var argRoleNames = args.Select(x => x.Value).SkipLast(1);
+
+            foreach (var roleName in argRoleNames)
             {
-                var command = new SetRoleAsUnsafeCommand(roleName, contexts.Server.Id);
-                await _commandBus.ExecuteAsync(command);
+                var serverRole = _usersRolesService.GetRoleByName(roleName, contexts.Server);
+
+                if (serverRole == null)
+                {
+                    await messageService.SendResponse(x => x.RoleNotFoundOrIsNotSafe(contexts, roleName), contexts);
+                    continue;
+                }
+
+                if (isSafe && !safeRoles.Select(x => x.Name).Where(x => x == roleName).Select(x => x).Any())
+                {
+                    await messageService.SendResponse(x => x.RoleIsSafeAlready(roleName), contexts);
+                    continue;
+                }
+                else if (!isSafe && !safeRoles.Select(x => x.Name).Where(x => x == roleName).Select(x => x).Any())
+                {
+                    await messageService.SendResponse(x => x.RoleIsUnsafeAlready(roleName), contexts);
+                    continue;
+                }
+
+                if (isSafe)
+                {
+                    await _commandBus.ExecuteAsync(new SetRoleAsSafeCommand(roleName, contexts.Server.Id));
+                }
+                else
+                {
+                    await _commandBus.ExecuteAsync(new SetRoleAsUnsafeCommand(roleName, contexts.Server.Id));
+                }
             }
 
-            var messagesService = _messagesServiceFactory.Create(contexts);
-            await messagesService.SendResponse(x => x.RoleSettingsChanged(roleName), contexts);
+            string msg = string.Join(", ",
+                args.Select(x => x.Value).SkipLast(1)) + " have been set as " + lastArgument;
```

Comment:
lepiej zrobić to z użyciem Repsonses, żeby było możliwe przetłumaczenie na inny język, lub styl innego serwera 

możliwe że tutaj będzie to wymagało przebudowy responses bo jeszcze nie obsługujemy tam list (jest w planach dodanie)
jeśli masz pomysł na zrobienie tego w obecnej wersji to dodaj
jak nie to możesz zostawić